iceoryx2-bb/lock-free/src/mpmc/bit_set.rs

```diff
@@ -213,23 +262,50 @@ pub mod details {
                 id
             );
 
-            let bitset_index = id / BITSET_ELEMENT_BITSIZE;
-            let bit = id % BITSET_ELEMENT_BITSIZE;
-            self.set_bit(bitset_index, bit)
+            self.set_bit(id)
+        }
+
+        /// Resets the next set bit and returns the bit index. If no bit was set it returns
+        /// [`None`].
+        pub fn reset_next(&self) -> Option<usize> {
+            self.verify_init("reset_next");
+            if self.active_bits.load(Ordering::Relaxed) == 0 {
+                return None;
+            }
+
+            let mut current_position = self.reset_position.load(Ordering::Relaxed);
+            for _ in 0..self.capacity {
+                current_position = (current_position + 1) % self.capacity;
+
+                if self.clear_bit(current_position) {
+                    self.reset_position
+                        .store(current_position, Ordering::Relaxed);
+                    return Some(current_position);
+                }
+            }
```

Comment:
```suggestion
            let current_position = self.reset_position.load(Ordering::Relaxed);
            for pos in (current_position..self.capacity).chain(0..current_position) {
                if self.clear_bit(pos) {
                    self.reset_position.store(pos + 1, Ordering::Relaxed);
                    return Some(pos);
                }
            }
```
This is simpler and more performant

Reply:
There is another optimization when `active_bits == 1`, especially when the bitset is large and always the same bit is set.